### PR TITLE
Unit Testing

### LINKED ISSUE #42: Unit Testing
Add Unit Tests to the test the API.

### PR DESCRIPTION
**Pull request by karelumair**
Added tests for all the APIs

closes #42

### CODE REVIEW

File: utils/scraping.py [hunk 1]
```diff
@@ -56,6 +58,8 @@ def scrape_quotes(self, url) -> dict:
         try:
             next_li = driver.find_element(By.CLASS_NAME, "next")
             next_page = True
+            if single_page:
+                next_page = False
```
Comment (pratikGhodke1) [inline utils/scraping.py]:
```suggestion
            next_page = False if single_page else True
```

File: constants/app_constants.py [hunk 2]
```diff
@@ -1,3 +0,0 @@
-"""Constants"""
-
-QUOTES_URL = "https://quotes.toscrape.com/"
```
Comment (pratikGhodke1) [inline constants/app_constants.py]: Actually don't delete this file, keep it as it is

File: utils/scraping.py [hunk 3]
```diff
@@ -11,9 +11,11 @@
 from utils.utils import init_driver, get_date
 from database.models import ScrapedAuthor, Author, Quote
 
+QUOTES_URL = "https://quotes.toscrape.com/"
+
```
Comment (pratikGhodke1) [inline utils/scraping.py]:
```suggestion
from constants.app_constants import QUOTES_URL
```
Use this `QUOTES_URL` below

File: tests/quote_test.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+import unittest
+from app import create_app
+from database.models import Quote, Author
+
+
+class QuoteTest(unittest.TestCase):
+    """Quote Test Class"""
+
+    def setUp(self):
```
Comment (pratikGhodke1) [inline tests/quote_test.py:11]:
```suggestion
    def set_up(self):
```
Or is it must to use `setup`?

File: tests/quote_test.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+    def test_update_quote(self):
+        """Test for updating Quote"""
+
+        data = {"quote": "The Quote is Updated..."}
+        response = self.client.put(f"/quotes/{self.quote_id}/", json=data)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(Quote.objects.get(id=self.quote_id).quote, data["quote"])
+
+    def test_pydantic_update_quote(self):
```
Comment (pratikGhodke1) [inline tests/quote_test.py]:
```suggestion
    def testupdate_quote_request_validation(self):
```

File: tests/author_test.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+import unittest
+from app import create_app
+from database.models import Author
+
+
+class AuthorTest(unittest.TestCase):
+    """Author Test Class"""
+
+    def setUp(self):
```
Comment (pratikGhodke1) [inline tests/author_test.py:11]:
```suggestion
    def set_up(self):
```

File: tests/author_test.py [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+    def test_update_author(self):
+        """Test for updating Author"""
+
+        data = {"name": "User Test"}
+        response = self.client.put(f"/authors/{self.author_id}/", json=data)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(Author.objects.get(id=self.author_id).name, data["name"])
+
+    def test_pydantic_update_author(self):
```
Comment (pratikGhodke1) [inline tests/author_test.py]:
```suggestion
    def test_update_author_request_validation(self):
```

**Review by pratikGhodke1 — CHANGES_REQUESTED** [review]
Make suggested changes

File: tests/author_test.py (hunk 6 above)
Comment (karelumair) [reply tests/author_test.py:11]: This leads to an error. unittest uses "setUp" method to setup for each test case.

File: tests/quote_test.py (hunk 4 above)
Comment (karelumair) [reply tests/quote_test.py:11]: yes it must be setUp

Comment (pratikGhodke1) [reply tests/quote_test.py:11]: Okay, there must be a `tearDown` method which can destroy things that are being created in `setUp` method. Can you check that and add it?

**Pull request merged**